source/class/qx/tool/compiler/app/Cldr.js

```diff
@@ -564,29 +580,28 @@ qx.Class.define("qx.tool.compiler.app.Cldr", {
             });
 
             var monthContext = get("months[0].monthContext", cal);
-            find(monthContext, "type", "format", function (row) {
-              find(row.monthWidth, "type", "abbreviated", function (row) {
-                for (var i = 0; i < row.month.length; i++) {
-                  var m = row.month[i];
-                  cldr["cldr_month_format_abbreviated_" + m["$"].type] =
-                    getText(m);
-                }
+
+            const parseMonth = (months, where) => {
+              if (!months){
+                return;
+              }
+              months.forEach(month => {
+                cldr[where + "_" + month["$"].type] = getText(month);
               });
-            });
-            find(monthContext, "type", "format", function (row) {
-              find(row.monthWidth, "type", "wide", function (row) {
-                for (var i = 0; i < row.month.length; i++) {
-                  var m = row.month[i];
-                  cldr["cldr_month_format_wide_" + m["$"].type] = getText(m);
-                }
+            }
+
+            const parseMonthContext = (what, where) => {
+              find(monthContext, "type", "format", function (row) {
+                find(row.monthWidth, "type", what, function (row) {
+                  parseMonth(row.month, where);
+                });
               });
-            });
```

Comment:
This change looks like it's lost the effect of `cldr["cldr_month_format_wide_" + m["$"].type] = getText(m);`. Is that no longer needed in this version?

Reply:
I replaced 2 code duplication by a function. There are two calls of the function for wide and abbreviated:
```
 parseMonthContext("abbreviated","cldr_month_format_abbreviated");
 parseMonthContext("wide","cldr_month_format_wide");
 ```
 Didn't understand a problem.

Reply:
I see what you did. Sorry. Yes, should be fine.

Reply:
I refactor this duplication bc I had to add the same check 3 times.